### ISSUE TITLE
RFC 561 work to improve our drilldown features

### ISSUE THREAD

**Issue by Joelkw** [sourcegraph/sourcegraph#32710]
It would be great to move forward with the needs described in [RFC 561](https://docs.google.com/document/d/1kgBu3aWy0XMJYjW0lwRTmVm37fwbqU-EMwIzfUWtYHw/edit#) because these tie into our use case flows for FY23 – from an insight, being able to see changes helps you fix vulnerabilities or follow up with whoever may have improved/worsened code health. I have added some relevant user quotes to the RFC. 

I would like us to make sure this doesn't fall through the cracks and think we should track it in our iteration. A goal for this work would be if we end this issue with proposals that fit into the following categories: 

1. Iterations we could do to improve the drilldown experience now, based on our existing architecture
   2. This includes purely frontend things, like displaying information on the page that is easy for us to fetch already! 
3. Things we would like to do but would not be possible to do right now without other foundational work, and the longer-term implications of drilldown features we _can't_ do now, but would want to explore enabling – what foundational work would be required

Think of this as our normal user/design brief RFC process, but as an experiment and because engineers were excited about it, we're starting with some of the product/technical side rather than the product/design side.

**Comment by coury-clark** [comment]
@vovakulikov I've added all of the content I had prepared, do you have anything else to add? I think there is more that can be done here, but might be worth exploring when this comes up on our roadmap?

**Comment by vovakulikov** [comment]
@coury-clark Agreed. Let's go one step at a time. And other further ideas kind of rely on current/your ideas in RFC so we will see what we can do there later.
